libarchive/archive_read_support_format_7zip.c

```diff
@@ -1063,6 +1063,9 @@ ppmd_read(void *p)
 		ssize_t bytes_avail = 0;
 		const uint8_t* data = __archive_read_ahead(a,
 		    (size_t)zip->ppstream.stream_in+1, &bytes_avail);
+		if(data == NULL) {
```

Comment:
I think this should be combined into the following `if` so that the error is correctly reported, not just ignored.

That is, change the following `if` to something like
```
  if (data == NULL || bytes_avail < zip->ppstream.stream_in+1) {
```